design/rest-api-guideline.md

```diff
@@ -454,7 +454,29 @@ Keycloak does not provide rate limiting capabilities and for such it is necessar
 
 ### Filtering
 
-TODO
+Sometimes, we need more control over the content that will return from the server, so that only a subset of the data is available in the response. For example, a list user's groups or permissions. 
+
+The Keycloak admin REST API allows to search for users based on `e-mail`, `first name`, `lastname` and other optional parameters. However, such implementation is limited to a particular context (e.g, collection of users).
+
+Filtering can be implemented as a query parameter, like we already did for the current REST API, with some enhancements. To filter on a field, should be as simple as the following request:
+
+```
+/{realm}/apis/admin/users/v1?group=users1&realm=master&sort=desc
+```
+
+Multiple filters should result in an implicit `AND`, so in our example `/v1?group=users1&role=role1&realm=master&sort=desc` would provide results based on `groups`, `roles` and `realm`. Now let's say we want filter based on more than one groups:
```

Comment:
As a guideline, I think you should consider how you defined `AND` to define that `clients should use the list of attributes supported by the API to filter results accordingly`.

Reply:
The definitions here are really about: